fdbserver/storageserver.actor.cpp

```diff
@@ -1946,12 +2005,44 @@ ACTOR Future<Void> logFetchKeysWarning(AddingShard* shard) {
 	}
 }
 
+class FetchKeysMetricReporter {
+	const UID uid;
+	const double startTime;
+	int fetchedBytes;
+	StorageServer::FetchKeysHistograms& histograms;
+	StorageServer::CurrentRunningFetchKeys& currentRunning;
+
+public:
+	FetchKeysMetricReporter(const UID& uid_, const double startTime_, const KeyRange& keyRange, StorageServer::FetchKeysHistograms& histograms_, StorageServer::CurrentRunningFetchKeys& currentRunning_)
+		 : uid(uid_), startTime(startTime_), fetchedBytes(0), histograms(histograms_), currentRunning(currentRunning_) {
+
+		currentRunning.recordStart(uid, keyRange);
+	}
+
+	void addFetchedBytes(const int bytes) { fetchedBytes += bytes; }
+
+	~FetchKeysMetricReporter() {
+		const double latency = now() - startTime;
+		if (latency == 0) return;
```

Comment:
We should still sample the histogram if the latency is zero.  It could only be zero if fetch keys returns synchronously, which shouldn't happen.  To avoid the divide by zero, set it to 1nsec for the purposes of the bandwidth computation -- that'll cause a sample in the top bucket of the histogram, which is a reasonable behavior.